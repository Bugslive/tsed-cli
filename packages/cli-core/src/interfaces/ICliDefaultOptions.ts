export interface ICliDefaultOptions {
  verbose: boolean;
}
