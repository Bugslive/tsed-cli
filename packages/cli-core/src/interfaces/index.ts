import {Type} from "@tsed/core";
import {CommandProvider} from "./CommandProvider";
import {PackageJson} from "./PackageJson";

export * from "./CommandProvider";
export * from "./CommandParameters";
export * from "./CommandMetadata";
export * from "./CliDefaultOptions";
export * from "./PackageJson";
export * from "./Tasks";

// eslint-disable-next-line @typescript-eslint/no-namespace
declare namespace TsED {
  interface Configuration {
    /**
     *
     */
    name: string;
    /**
     *
     */
    commands: Type<CommandProvider>[];
    /**
     *
     */
    pkg: PackageJson;
    /**
     *
     */
    templateDir: string;
    /**
     *
     */
    project?: {
      root?: string;
      srcDir?: string;
    };
  }
}
