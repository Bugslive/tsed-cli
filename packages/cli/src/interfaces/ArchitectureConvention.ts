export enum ArchitectureConvention {
  DEFAULT = "arc_default",
  FEATURE = "feature"
}

declare global {
  namespace TsED {
    interface ProjectPreferences {
      architecture: ArchitectureConvention;
    }
  }
}
