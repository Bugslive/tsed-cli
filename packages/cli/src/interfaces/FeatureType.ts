export enum FeatureType {
  GRAPHQL = "graphql",
  SOCKETIO = "socketio",
  SWAGGER = "swagger",
  OIDC = "oidc",
  PASSPORTJS = "passportjs",
  COMMANDS = "commands",
  DB = "db",

  // ORM
  PRISMA = "prisma",

  MONGOOSE = "mongoose",

  // TYPEORM
  TYPEORM = "typeorm",
  TYPEORM_MYSQL = "typeorm:mysql",
  TYPEORM_MARIADB = "typeorm:mariadb",
  TYPEORM_POSTGRES = "typeorm:postgres",
  TYPEORM_COCKROACHDB = "typeorm:cockroachdb",
  TYPEORM_SQLITE = "typeorm:sqlite",
  TYPEORM_BETTER_SQLITE3 = "typeorm:better-sqlite3",
  TYPEORM_CORDOVA = "typeorm:cordova",
  TYPEORM_NATIVESCRIPT = "typeorm:nativescript",
  TYPEORM_ORACLE = "typeorm:oracle",
  TYPEORM_MSSQL = "typeorm:mssql",
  TYPEORM_MONGODB = "typeorm:mongodb",
  TYPEORM_SQLJS = "typeorm:sqljs",
  TYPEORM_REACTNATIVE = "typeorm:reactnative",
  TYPEORM_EXPO = "typeorm:expo",

  // TESTING
  TESTING = "testing",
  JEST = "jest",
  MOCHA = "mocha",
  LINTER = "linter",
  ESLINT = "eslint",
  LINT_STAGED = "lintstaged",
  PRETTIER = "prettier",

  // BUNDLER
  BUNDLER = "bundler",
  BABEL = "babel",
  WEBPACK = "babel:webpack"
}
