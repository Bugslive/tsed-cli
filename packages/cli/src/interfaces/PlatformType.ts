export enum PlatformType {
  EXPRESS = "express",
  KOA = "koa"
}
