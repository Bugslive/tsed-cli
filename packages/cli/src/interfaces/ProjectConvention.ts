export enum ProjectConvention {
  DEFAULT = "conv_default",
  ANGULAR = "angular"
}

declare global {
  namespace TsED {
    interface ProjectPreferences {
      convention: ProjectConvention;
    }
  }
}
