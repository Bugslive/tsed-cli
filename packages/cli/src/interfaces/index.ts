export * from "./ArchitectureConvention.js";
export * from "./PlatformType.js";
export * from "./ProjectConvention.js";
