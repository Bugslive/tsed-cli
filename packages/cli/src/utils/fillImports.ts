import {ArchitectureConvention, PlatformType} from "../interfaces";

export function fillImports(ctx: any) {
  ctx = {...ctx};
  Object.values(PlatformType).forEach((type) => {
    ctx[type] = ctx.platform === type;
  });

  const isFeature = ctx.architecture === ArchitectureConvention.FEATURE;

  ctx.barrels = JSON.stringify(
    [
      isFeature ? "./src/rest" : "./src/controllers/rest",
      (ctx.swagger || ctx.oidc) && (isFeature ? "./src/pages" : "./src/controllers/pages"),
      ctx.oidc && "./src/interactions",
      ctx.graphql && "./src/datasources",
      "./src/resolvers"
    ].filter(Boolean)
  );

  ctx.imports = [
    ctx.express && {from: "@tsed/platform-express", comment: " // /!\\ keep this import"},
    ctx.express && {symbols: "bodyParser", from: "body-parser"},
    ctx.express && {symbols: "compress", from: "compression"},
    ctx.express && {symbols: "cookieParser", from: "cookie-parser"},
    ctx.express && {symbols: "methodOverride", from: "method-override"},
    ctx.express && {symbols: "cors", from: "cors"},
    ctx.koa && {from: "@tsed/platform-koa", comment: " // /!\\ keep this import"},
    ctx.koa && {symbols: "bodyParser", from: "koa-bodyparser"},
    ctx.koa && {symbols: "compress", from: "koa-compress"},
    ctx.koa && {symbols: "cors", from: "@koa/cors"},
    ctx.koa && {symbols: "methodOverride", from: "koa-override", tsIgnore: true},
    {from: "@tsed/ajv"},
    ctx.swagger && {from: "@tsed/swagger"},
    ctx.mongoose && {from: "@tsed/mongoose"},
    ctx.oidc && {from: "@tsed/oidc-provider"},
    ctx.graphql && {from: "@tsed/typegraphql"},
    ctx.graphql && {from: "./datasources/index"},
    ctx.graphql && {from: "./resolvers/index"},
    {symbols: "{config}", from: "./config/index"},
    {symbols: "* as rest", from: isFeature ? "./rest/index" : "./controllers/rest/index"},
    (ctx.swagger || ctx.oidc) && {
      symbols: "* as pages",
      from: isFeature ? "./pages/index" : "./controllers/pages/index"
    },
    ctx.oidc && {symbols: "{InteractionsController}", from: "./controllers/oidc/InteractionsController"}
  ].filter(Boolean);

  return ctx;
}
