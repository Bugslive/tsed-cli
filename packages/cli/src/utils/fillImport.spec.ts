import {ArchitectureConvention} from "../interfaces";
import {fillImports} from "./fillImports";

describe("fillImports()", () => {
  it.each([
    {
      architecture: ArchitectureConvention.FEATURE,
      oidc: false,
      graphql: false,
      swagger: false,
      passportjs: false,
      express: false,
      koa: false,
      mongoose: false
    },
    {
      architecture: ArchitectureConvention.FEATURE,
      oidc: true,
      graphql: false,
      swagger: false,
      passportjs: false,
      express: true,
      koa: false,
      mongoose: false
    },
    {
      architecture: ArchitectureConvention.DEFAULT,
      oidc: true,
      graphql: false,
      swagger: false,
      passportjs: false,
      express: false,
      koa: true,
      mongoose: false
    },
    {
      architecture: ArchitectureConvention.FEATURE,
      oidc: false,
      graphql: false,
      swagger: true,
      passportjs: false,
      express: false,
      koa: false,
      mongoose: false
    },
    {
      architecture: ArchitectureConvention.DEFAULT,
      oidc: false,
      graphql: false,
      swagger: true,
      passportjs: false,
      express: false,
      koa: false,
      mongoose: false
    },
    {
      architecture: ArchitectureConvention.FEATURE,
      oidc: true,
      graphql: true,
      swagger: true,
      passportjs: true,
      express: true,
      koa: false,
      mongoose: true
    },
    {
      architecture: ArchitectureConvention.DEFAULT,
      oidc: true,
      graphql: true,
      swagger: true,
      passportjs: true,
      express: false,
      koa: true,
      mongoose: true
    }
  ])(
    "it should return barrels and imports files for ($architecture, oidc: $oidc, graphql: $graphql, swagger: $swagger, passportjs: $passportjs, express: $express, koa: $koa, mongoose: $mongoose)",
    (ctx) => {
      expect(fillImports(ctx)).toMatchSnapshot();
    }
  );
});
