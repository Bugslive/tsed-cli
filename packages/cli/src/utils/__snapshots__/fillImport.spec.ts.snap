// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fillImports() > should return barrels and imports files for ('arc_default', oidc: false, graphql: false, swagger: true, passportjs: false, express: false, koa: false, mongoose: false) 1`] = `
{
  "architecture": "arc_default",
  "barrels": "["./src/controllers/rest/index.js","./src/controllers/pages/index.js"]",
  "express": false,
  "graphql": false,
  "imports": [
    {
      "from": "@tsed/ajv",
    },
    {
      "from": "@tsed/swagger",
    },
    {
      "from": "./config/index.js",
      "symbols": "{config}",
    },
    {
      "from": "./controllers/rest/index.js",
      "symbols": "* as rest",
    },
    {
      "from": "./controllers/pages/index.js",
      "symbols": "* as pages",
    },
  ],
  "koa": false,
  "mongoose": false,
  "oidc": false,
  "passportjs": false,
  "swagger": true,
}
`;

exports[`fillImports() > should return barrels and imports files for ('arc_default', oidc: true, graphql: false, swagger: false, passportjs: false, express: false, koa: true, mongoose: false) 1`] = `
{
  "architecture": "arc_default",
  "barrels": "["./src/controllers/rest/index.js","./src/interactions/index.js"]",
  "express": false,
  "graphql": false,
  "imports": [
    {
      "from": "@tsed/ajv",
    },
    {
      "from": "@tsed/oidc-provider",
    },
    {
      "from": "./config/index.js",
      "symbols": "{config}",
    },
    {
      "from": "./controllers/rest/index.js",
      "symbols": "* as rest",
    },
    {
      "from": "./controllers/pages/index.js",
      "symbols": "* as pages",
    },
    {
      "from": "./controllers/oidc/InteractionsController.js",
      "symbols": "{InteractionsController}",
    },
  ],
  "koa": false,
  "mongoose": false,
  "oidc": true,
  "passportjs": false,
  "swagger": false,
}
`;

exports[`fillImports() > should return barrels and imports files for ('arc_default', oidc: true, graphql: true, swagger: true, passportjs: true, express: false, koa: true, mongoose: true) 1`] = `
{
  "architecture": "arc_default",
  "barrels": "["./src/controllers/rest/index.js","./src/controllers/pages/index.js","./src/interactions/index.js","./src/datasources/index.js","./src/resolvers/index.js"]",
  "express": false,
  "graphql": true,
  "imports": [
    {
      "from": "@tsed/ajv",
    },
    {
      "from": "@tsed/swagger",
    },
    {
      "from": "@tsed/mongoose",
    },
    {
      "from": "@tsed/oidc-provider",
    },
    {
      "from": "@tsed/passport",
    },
    {
      "from": "@tsed/typegraphql",
    },
    {
      "from": "./datasources/index.js",
    },
    {
      "from": "./resolvers/index.js",
    },
    {
      "from": "./config/index.js",
      "symbols": "{config}",
    },
    {
      "from": "./controllers/rest/index.js",
      "symbols": "* as rest",
    },
    {
      "from": "./controllers/pages/index.js",
      "symbols": "* as pages",
    },
    {
      "from": "./controllers/oidc/InteractionsController.js",
      "symbols": "{InteractionsController}",
    },
  ],
  "koa": false,
  "mongoose": true,
  "oidc": true,
  "passportjs": true,
  "swagger": true,
}
`;

exports[`fillImports() > should return barrels and imports files for ('feature', oidc: false, graphql: false, swagger: false, passportjs: false, express: false, koa: false, mongoose: false) 1`] = `
{
  "architecture": "feature",
  "barrels": "["./src/rest/index.js"]",
  "express": false,
  "graphql": false,
  "imports": [
    {
      "from": "@tsed/ajv",
    },
    {
      "from": "./config/index.js",
      "symbols": "{config}",
    },
    {
      "from": "./rest/index.js",
      "symbols": "* as rest",
    },
  ],
  "koa": false,
  "mongoose": false,
  "oidc": false,
  "passportjs": false,
  "swagger": false,
}
`;

exports[`fillImports() > should return barrels and imports files for ('feature', oidc: false, graphql: false, swagger: true, passportjs: false, express: false, koa: false, mongoose: false) 1`] = `
{
  "architecture": "feature",
  "barrels": "["./src/rest/index.js","./src/pages/index.js"]",
  "express": false,
  "graphql": false,
  "imports": [
    {
      "from": "@tsed/ajv",
    },
    {
      "from": "@tsed/swagger",
    },
    {
      "from": "./config/index.js",
      "symbols": "{config}",
    },
    {
      "from": "./rest/index.js",
      "symbols": "* as rest",
    },
    {
      "from": "./pages/index.js",
      "symbols": "* as pages",
    },
  ],
  "koa": false,
  "mongoose": false,
  "oidc": false,
  "passportjs": false,
  "swagger": true,
}
`;

exports[`fillImports() > should return barrels and imports files for ('feature', oidc: true, graphql: false, swagger: false, passportjs: false, express: true, koa: false, mongoose: false) 1`] = `
{
  "architecture": "feature",
  "barrels": "["./src/rest/index.js","./src/interactions/index.js"]",
  "express": false,
  "graphql": false,
  "imports": [
    {
      "from": "@tsed/ajv",
    },
    {
      "from": "@tsed/oidc-provider",
    },
    {
      "from": "./config/index.js",
      "symbols": "{config}",
    },
    {
      "from": "./rest/index.js",
      "symbols": "* as rest",
    },
    {
      "from": "./pages/index.js",
      "symbols": "* as pages",
    },
    {
      "from": "./controllers/oidc/InteractionsController.js",
      "symbols": "{InteractionsController}",
    },
  ],
  "koa": false,
  "mongoose": false,
  "oidc": true,
  "passportjs": false,
  "swagger": false,
}
`;

exports[`fillImports() > should return barrels and imports files for ('feature', oidc: true, graphql: true, swagger: true, passportjs: true, express: true, koa: false, mongoose: true) 1`] = `
{
  "architecture": "feature",
  "barrels": "["./src/rest/index.js","./src/pages/index.js","./src/interactions/index.js","./src/datasources/index.js","./src/resolvers/index.js"]",
  "express": false,
  "graphql": true,
  "imports": [
    {
      "from": "@tsed/ajv",
    },
    {
      "from": "@tsed/swagger",
    },
    {
      "from": "@tsed/mongoose",
    },
    {
      "from": "@tsed/oidc-provider",
    },
    {
      "from": "@tsed/passport",
    },
    {
      "from": "@tsed/typegraphql",
    },
    {
      "from": "./datasources/index.js",
    },
    {
      "from": "./resolvers/index.js",
    },
    {
      "from": "./config/index.js",
      "symbols": "{config}",
    },
    {
      "from": "./rest/index.js",
      "symbols": "* as rest",
    },
    {
      "from": "./pages/index.js",
      "symbols": "* as pages",
    },
    {
      "from": "./controllers/oidc/InteractionsController.js",
      "symbols": "{InteractionsController}",
    },
  ],
  "koa": false,
  "mongoose": true,
  "oidc": true,
  "passportjs": true,
  "swagger": true,
}
`;
